moveit_experimental/collision_distance_field/src/collision_robot_distance_field.cpp

```diff
@@ -524,446 +700,671 @@ bool CollisionRobotDistanceField::getIntraGroupProximityGradients(boost::shared_
+    for (unsigned int j = 0; j < sphere_representation->getCollisionSpheres().size(); j++)
+    {
+      tf::pointEigenToMsg(sphere_representation->getSphereCenters()[j], sphere_marker.pose.position);
+      sphere_marker.scale.x = sphere_marker.scale.y = sphere_marker.scale.z =
+          2 * sphere_representation->getCollisionSpheres()[j].radius_;
+      sphere_marker.id = id;
+      id++;
+
+      model_markers.markers.push_back(sphere_marker);
+    }
   }
 }
 
-void CollisionRobotDistanceField::addLinkBodyDecompositions(double resolution,
-                                                            const std::map<std::string, std::vector<CollisionSphere> >& link_spheres) 
+void CollisionRobotDistanceField::addLinkBodyDecompositions(
+    double resolution, const std::map<std::string, std::vector<CollisionSphere>> &link_spheres)
 {
-  const std::vector<robot_model::LinkModel*>& link_models = kmodel_->getLinkModelsWithCollisionGeometry();
-  for(unsigned int i = 0; i < link_models.size(); i++) {
-    if(!link_models[i]->getShape()) {
-      logWarn("No collision geometry for link model %s though there should be", link_models[i]->getName().c_str());
+  ROS_ASSERT_MSG(robot_model_, "RobotModelPtr is invalid");
+  const std::vector<const moveit::core::LinkModel *> &link_models = robot_model_->getLinkModelsWithCollisionGeometry();
+
+  for (unsigned int i = 0; i < link_models.size(); i++)
+  {
+    if (link_models[i]->getShapes().empty())
+    {
+      ROS_INFO_STREAM("Skipping model generation for link " << link_models[i]->getName() << " since it contains no "
+                                                                                            "geometries");
       continue;
     }
-    logDebug("Generating model for %s", link_models[i]->getName().c_str());
-    BodyDecompositionPtr bd(new BodyDecomposition(link_models[i]->getShape(), resolution, resolution));
-    if(link_spheres.find(link_models[i]->getName()) != link_spheres.end()) {
+
+    BodyDecompositionPtr bd(new BodyDecomposition(link_models[i]->getShapes(),
+                                                  link_models[i]->getCollisionOriginTransforms(), resolution,
+                                                  getLinkPadding(link_models[i]->getName())));
+
+    ROS_INFO("Generated model for %s", link_models[i]->getName().c_str());
+
+    if (link_spheres.find(link_models[i]->getName()) != link_spheres.end())
+    {
       bd->replaceCollisionSpheres(link_spheres.find(link_models[i]->getName())->second, Eigen::Affine3d::Identity());
     }
     link_body_decomposition_vector_.push_back(bd);
-    link_body_decomposition_index_map_[link_models[i]->getName()] = link_body_decomposition_vector_.size()-1;
+    link_body_decomposition_index_map_[link_models[i]->getName()] = link_body_decomposition_vector_.size() - 1;
   }
+
+  ROS_INFO("The sizes are %d, %d", link_body_decomposition_vector_.size(), link_body_decomposition_index_map_.size());
```

Comment:
according to travis: warning: format ‘%d’ expects argument of type ‘int’, but argument 8 has type ‘std::vector<boost::shared_ptr<const collision_detection::BodyDecomposition> >::size_type {aka long unsigned int}’